packages/eslint-plugin/tests/rules/no-misused-promises.test.ts

```diff
@@ -970,5 +970,29 @@ console.log({ ...(condition ? Promise.resolve({ key: 42 }) : {}) });
         { line: 7, messageId: 'spread' },
       ],
     },
+    {
+      code: `
+type MyUnion = (() => void) | boolean;
+
+function restPromises(first: Boolean, ...callbacks: Array<() => void>): void {}
+function restUnion(first: string, ...callbacks: Array<MyUnion>): void {}
+
+restPromises(
+  true,
+  () => Promise.resolve(true),
+  () => Promise.resolve(null),
+  () => true,
+  () => Promise.resolve('Hello'),
+);
+
+restUnion('Testing', false, () => Promise.resolve(true));
+      `,
+      errors: [
+        { line: 9, messageId: 'voidReturnArgument' },
+        { line: 10, messageId: 'voidReturnArgument' },
+        { line: 12, messageId: 'voidReturnArgument' },
+        { line: 15, messageId: 'voidReturnArgument' },
+      ],
```

Comment:
We generally try to keep to one type of test case per actual test case. This looks like it tests both a `restPromises` function and a `restUnion` function. Could you please split these into two cases?

Reply:
I've split out the union code into a separate test case